src/main/python/code_style/rules/naming_style.py

```diff
@@ -0,0 +1,44 @@
+
+from functools import reduce
+from nltk.corpus import words
+import re
+
+try:
+    import xml.etree.cElementTree as XML
+except ImportError:
+    import xml.etree.ElementTree as XML
+
+def is_word(s):
+    return s.lower() in words.words()
+
+def is_underscore_naming_style(s):
+    return re.search('_', s)
+
+def split_by_uppercase(x, y):
```

Comment:
add a doc since it's hard to understand
```python
def split_by_uppercase(x, y):
    """
    Usage:
        words = reduce(split_by_uppercase, 'theBigCat', [''])
        assert words == ['the', 'big', 'cat']
    """
    if y.isupper():
        return x + [y.lower()]
    else:
        x[-1] += y
        return x
```